src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -26,7 +26,9 @@ internal sealed class Http3RequestStream : IHttpStreamHeadersHandler, IAsyncDisp
         private Http3Connection _connection;
         private long _streamId = -1; // A stream does not have an ID until the first I/O against it. This gets set almost immediately following construction.
         private readonly QuicStream _stream;
+        private volatile bool _concurrentWrite;
         private ArrayBuffer _sendBuffer;
+        private volatile bool _concurrentRead;
```

Comment:
Nit: Are there any reordering concerns here such that we need volatile?
Or is it more for documentation purposes that there are multiple threads involved?

Reply:
Not reordering, just making it more probable that the other threat will see the value change. If I'm not mistaken, if the variable is **NOT** volatile, the other threat might never see the change:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile#example
>However, without that modifier on the _shouldStop member, the behavior is unpredictable. The DoWork method may optimize the member access, resulting in reading stale data.

Reply:
Thanks for the detailed explanation!

I'll keep it as-is, while trying to retain the info you shared in my brain :smile: